example/tests/test_serializers.py

```diff
@@ -30,6 +47,59 @@ def setUp(self):
                 Author.objects.create(name=name, email='{}@example.org'.format(name))
             )
 
+    def test_forward_relationship_not_loaded_when_not_included(self):
+        to_representation_method = 'example.serializers.BlogSerializer.to_representation'
+        with mock.patch(to_representation_method) as mocked_serializer:
+            class EntrySerializer(ModelSerializer):
+                blog = BlogSerializer()
+
+                class Meta:
+                    model = Entry
+                    fields = '__all__'
+
+            request_without_includes = Request(request_factory.get('/'))
+            serializer = EntrySerializer(context={'request': request_without_includes})
+            serializer.to_representation(self.entry)
+
+            mocked_serializer.assert_not_called()
+
+    def test_forward_relationship_optimization_correct_representation(self):
+        class EntrySerializer(ModelSerializer):
+            blog = BlogSerializer()
+
+            class Meta:
+                model = Entry
+                fields = '__all__'
+
+        request_without_includes = Request(request_factory.get('/'))
+        serializer = EntrySerializer(context={'request': request_without_includes})
+        result = serializer.to_representation(self.entry)
+
+        # Remove non deterministic fields
+        result.pop('created_at')
+        result.pop('modified_at')
+
+        expected = OrderedDict(
```

Comment:
ordering is not relevant in an assertion so I think the code would be more readable to use normal dict.

Reply:
Makes sense. I will also change it to assertDictEqual.